### PR TITLE
Remove URI validation when `@TestHTTPResource` is injected as String

### LINKED ISSUE #34939: Support `@TestHTTPResource` with path params
### Description

It would be nice if `@TestHTTPResource` (as `String`) supports path params, as RESTassured already supports it.
Currently, the following test throws an exception:

```
@QuarkusTest
public class UserTests {
    
    @TestHTTPEndpoint(UserResource.class)
    @TestHTTPResource("{userId}/order/{orderId}")
    String getUserOrderUrl;
    
    @Test
    void testGettingUserOrder() {
        int userId = 123;
        int orderId = 456;
        given().when().get(getUserOrderUrl, userId, orderId)
               .then().statusCode(200);
    }
}
```

```
Caused by:
    java.net.URISyntaxException: Illegal character in path at index 30: http://localhost:8081/user/v1/{userId}/order/{orderId}
    at java.base/java.net.URI$Parser.fail(URI.java:2963)
    at java.base/java.net.URI$Parser.checkChars(URI.java:3134)
    at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3216)
    at java.base/java.net.URI$Parser.parse(URI.java:3164)
    at java.base/java.net.URI.<init>(URI.java:628)
    at io.quarkus.test.common.http.TestHTTPResourceManager.inject(TestHTTPResourceManager.java:105)
    ... 83 more
```

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Eng-Fouad**
Fixes #34939

### CODE REVIEW

**Comment by geoand** [comment]
Thanks for this.

We'll need some tests for this as well

**Comment by geoand** [comment]
You can't use `@QuarkusTest` there. 

You can only use `@QuarkusTest` in our `integration-tests`.

The alternative is to use `QuarkusUnitTest`

**Comment by Eng-Fouad** [comment]
> You can't use `@QuarkusTest` there.
> 
> You can only use `@QuarkusTest` in our `integration-tests`.
> 
> The alternative is to use `QuarkusUnitTest`

I tried `QuarkusUnitTest` before that, but I got this exception:

```
@RegisterExtension
static final QuarkusUnitTest config = new QuarkusUnitTest().withApplicationRoot((jar) -> jar.addClass(UserResource.class));
```

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.977 s <<< FAILURE! -- in io.quarkus.test.junit.http.TestHTTPResourcePathParamTest
[ERROR] io.quarkus.test.junit.http.TestHTTPResourcePathParamTest -- Time elapsed: 2.977 s <<< ERROR!
java.lang.RuntimeException: org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:704)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
	Suppressed: java.lang.NoClassDefFoundError: io/quarkus/test/common/GroovyCacheCleaner
		at io.quarkus.test.QuarkusUnitTest.afterAll(QuarkusUnitTest.java:762)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$18(ClassBasedTestDescriptor.java:461)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$19(ClassBasedTestDescriptor.java:461)
		at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeAfterAllCallbacks(ClassBasedTestDescriptor.java:461)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.after(ClassBasedTestDescriptor.java:235)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.after(ClassBasedTestDescriptor.java:84)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:161)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:161)
		... 35 more
	Caused by: java.lang.ClassNotFoundException: io.quarkus.test.common.GroovyCacheCleaner
		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
		at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
		... 46 more
Caused by: org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:676)
	... 43 more
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.runner.bootstrap.RunningQuarkusApplicationImpl.instance(RunningQuarkusApplicationImpl.java:90)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:668)
	... 43 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:119)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.quarkus.runner.bootstrap.RunningQuarkusApplicationImpl.instance(RunningQuarkusApplicationImpl.java:85)
	... 44 more
Caused by: java.lang.IllegalStateException: Unable to locate CDIProvider
	at jakarta.enterprise.inject.spi.CDI.findAllProviders(CDI.java:139)
	at jakarta.enterprise.inject.spi.CDI.getCDIProvider(CDI.java:92)
	at jakarta.enterprise.inject.spi.CDI.current(CDI.java:65)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	... 46 more
```

**Comment by geoand** [comment]
It doesn't work inside the `quarkus-junit5`. You'll need add the test in a `deployment` module of one of the extensions that already uses `@TestHTTPResource` (like `quarkus-vertx-http-deployment`)

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by Eng-Fouad** [comment]
CI fails because of imports order. Force-pushed a new commit. @geoand Can you please start CI again? Thanks.

File: test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java [hunk 1]
```diff
@@ -102,7 +101,7 @@ public static void inject(Object testCase, List<Function<Class<?>, String>> endp
                     }
                     f.setAccessible(true);
                     try {
-                        f.set(testCase, provider.provide(new URI(val), f));
+                        f.set(testCase, provider.provide(val, f));
```
Comment (gastaldi) [inline test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java:104]: Won't this break tests expecting the `URI` type?

Comment (geoand) [reply test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java:104]: We have plenty of tests that use URI and they work (?) So it shouldn't be an issue

**Pull request merged**